hackathon_site/dashboard/frontend/src/pages/Teams/Teams.test.tsx

```diff
@@ -1,13 +1,54 @@
 import React from "react";
 
-import { render } from "testing/utils";
-import { teamsList } from "testing/mockData";
-import Teams from "pages/Teams/Teams";
-
-test("renders teams without crashing", () => {
-    const teamCodes = teamsList.map((team) => team.TeamName);
-    const { getByText } = render(<Teams />);
-    teamCodes.forEach((teamCode) => {
-        expect(getByText("Team " + teamCode)).toBeInTheDocument();
+import { makeMockApiListResponse, render } from "testing/utils";
+import { get } from "api/api";
+import { mockTeams } from "../../testing/mockData";
+import { waitFor, when } from "../../testing/utils";
+import Teams from "./Teams";
+import { queryByTestId } from "@testing-library/react";
+
+jest.mock("api/api", () => ({
+    ...jest.requireActual("api/api"),
+    get: jest.fn(),
+}));
+const mockedGet = get as jest.MockedFunction<typeof get>;
+
+const teamsUri = "/api/event/teams/";
+
+describe("Teams Page", () => {
+    it("Renders the teams", async () => {
+        const teamsApiResponse = makeMockApiListResponse(mockTeams);
+
+        when(mockedGet)
+            .calledWith(teamsUri, { limit: 24 })
```

Comment:
for all instances of limit: 24 just make sure to use a constant instead of 24

Reply:
done